core/src/main/java/org/apache/accumulo/core/client/Connector.java

```diff
@@ -237,4 +282,271 @@ public abstract BatchDeleter createBatchDeleter(String tableName, Authorizations
    * @since 1.7.0
    */
   public abstract ReplicationOperations replicationOperations();
+
+  /**
+   * Builds ConnectionInfo after all options have been specified
+   *
+   * @since 2.0.0
+   */
+  public interface ConnInfoFactory {
+
+    /**
+     * Builds ConnectionInfo after all options have been specified
+     *
+     * @return ConnectionInfo
+     */
+    ConnectionInfo info();
+  }
+
+  /**
+   * Builds Connector
+   *
+   * @since 2.0.0
+   */
+  public interface ConnectorFactory extends ConnInfoFactory {
+
+    /**
+     * Builds Connector after all options have been specified
+     *
+     * @return Connector
+     */
+    Connector build() throws AccumuloException, AccumuloSecurityException;
+
+  }
+
+  /**
+   * Builder method for setting Accumulo instance and zookeepers
+   *
+   * @since 2.0.0
+   */
+  public interface InstanceArgs {
+    AuthenticationArgs forInstance(String instanceName, String zookeepers);
+  }
+
+  /**
+   * Builder methods for creating Connector using properties
+   *
+   * @since 2.0.0
+   */
+  public interface PropertyOptions extends InstanceArgs {
+
+    /**
+     * Build using properties file
+     *
+     * @param propertiesFile
+     *          Path to properties file
+     * @return this builder
+     */
+    ConnectorFactory usingProperties(String propertiesFile);
```

Comment:
Some APIs will take a File (or in java 8 a Path) type for a method like this.  I find those more cumbersome to use (I seem to always have a string when going use something like this). From a design perspective I wonder if there is any advantage to this.  Seems like its only better when you already have a File object in that you get better type checking.

Reply:
I prefer using String too